### PR TITLE
Add BusinessId related methods to PatientRepository

### PR DESCRIPTION
**Pull request by vinaysshenoy**
Pivotal Story Link: https://www.pivotaltracker.com/story/show/164687599

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/patient/Patient.kt [hunk 1]
```diff
@@ -102,6 +102,14 @@ data class Patient(
     @Query("DELETE FROM patient")
     abstract fun clear()
 
+    @Query("""
+      SELECT P.* FROM Patient P
+        INNER JOIN BusinessId B ON B.patientUuid == P.uuid
+        WHERE B.identifier == :identifier AND B.deletedAt IS NULL
+        ORDER BY B.createdAt ASC
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/patient/Patient.kt]: Can you align these last three lines with the first one?

**Comment by saket** [comment]
@vinaysshenoy this PR looks good overall. I only have a couple of feedback related to formatting. Feel free to do them in your next PR if you want?

File: app/src/main/java/org/simple/clinic/patient/PatientRepository.kt [hunk 2]
```diff
@@ -449,4 +452,48 @@ class PatientRepository @Inject constructor(
...
+        .take(1)
+
+    val businessIdMetaAndVersionStream: Observable<Pair<String, BusinessId.MetaVersion>> =
+        Observables.combineLatest(currentUserStream, currentFacilityStream)
+            .map { (assignedByUser, assignedAtFacility) ->
+              BusinessIdMeta.BpPassportV1(assigningUserUuid = assignedByUser.uuid, assigningFacilityUuid = assignedAtFacility.uuid)
+            }
+            .map {
+              businessIdMetaAdapter.serialize(it, BusinessId.MetaVersion.BpPassportV1) to BusinessId.MetaVersion.BpPassportV1
+            }
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/patient/PatientRepository.kt]: Done.

File: app/src/main/java/org/simple/clinic/patient/Patient.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/patient/Patient.kt]: Done.

**Comment by vinaysshenoy** [comment]
> @vinaysshenoy this PR looks good overall. I only have a couple of feedback related to formatting. Feel free to do them in your next PR if you want?

@saket I have addressed feefback.

**Pull request merged**